go/processor/floria/precompiled.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"github.com/ethereum/go-ethereum/common"
+	geth "github.com/ethereum/go-ethereum/core/vm"
+)
+
+func handlePrecompiled(revision tosca.Revision, input tosca.Data, address tosca.Address, gas tosca.Gas) (tosca.CallResult, bool) {
+	contract, ok := precompiledContract(address, revision)
+	if !ok {
+		return tosca.CallResult{}, false
+	}
+	gasCost := contract.RequiredGas(input)
```

Comment:
documentation for this function says it does not need to check for overflow, since size of an input to be long enough to produce an overflow would already be too expensive. Regardless, I'm not happy about unchecked overflows, specially in non-performance-critical parts.